solr/solr-ref-guide/build.gradle

```diff
@@ -215,20 +216,21 @@ check.dependsOn checkLocalJavadocLinksSite, checkSite
 // Hook site building to assemble.
 assemble.dependsOn buildSite
 
+@CacheableTask
 abstract class PrepareSources extends DefaultTask {
     // Original Source files we'll be syncing <b>FROM</b>
     @InputDirectory
-    public DirectoryProperty srcDir = project.objects.directoryProperty()
+    abstract DirectoryProperty getSrcDir()
```

Comment:
Are these abstract properties automatically injected if you declare them like this? Where is the gradle docs that explains this (and which properties can be injected like so)? Thanks.

Reply:
Yes exactly, so the docs are [here](https://docs.gradle.org/current/userguide/lazy_configuration.html#working_with_files_in_lazy_properties) but afaik it does not explain the runtime magical injection happening.

Reply:
Ah, I see it now. Yeah - it'd be good to require the Inject annotation on those abstract properties too. This would be more intuitive (to me at least).